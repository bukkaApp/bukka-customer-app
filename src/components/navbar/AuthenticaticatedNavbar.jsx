import React, { Fragment, useState } from 'react';

import Brand from '../brand/Brand';
import CartIndicator from '../cart/CartIndicator';
import Cart, { CartDropdown } from '../cart/Cart';
import CartSection from '../cart/CartSection';
import UserDefaultImage from './common/UserDefaultImage';

import './navbar.scss';
import useClickOutside from '../../hooks/useClickOutside';

const AuthenticaticatedNavbar = () => {
  const wrapperRef = React.useRef(null);
  const [isFocused, setFocus] = useState(false);

  const handleClick = () => {
    setFocus(!isFocused);
  };

  useClickOutside(() => setFocus(false), wrapperRef);

  return (
    <Fragment>
      <nav className="container navbar navbar-light">
        <Brand />
        <div className="form-inline">
          <UserDefaultImage />
          <div className="position-relative">
            <div>
              <CartIndicator handleClick={handleClick} />
            </div>
            <div ref={wrapperRef}>
              <CartDropdown display={isFocused}>
                <Cart />
                <CartSection focus={isFocused} handleClick={handleClick} />
              </CartDropdown>
            </div>
          </div>
        </div>
      </nav>
    </Fragment>
  );
};
export default AuthenticaticatedNavbar;
