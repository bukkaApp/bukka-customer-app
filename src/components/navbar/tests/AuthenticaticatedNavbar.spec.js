import React from 'react';

import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import AuthenticaticatedNavbar from '../BukkaAuthenticatedNav';


describe('Primary Navbar', () => {
  const store = mockStore({
    navbarAuthReducer: { type: 'Sign Up' },
    authenticationReducer: {
      user: {},
      status: {
        authenticated: false,
        error: false
      },
      errorMessage: ''
    },
    deliveryModeReducer: {
      mode: 'pickup'
    }
  });

  const props = {
    status: { authenticated: true }
  };

  const wrapper = shallow(
    <Provider store={store}>
      <MemoryRouter>
        <AuthenticaticatedNavbar {...props} />
      </MemoryRouter>
    </Provider>
  );

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
