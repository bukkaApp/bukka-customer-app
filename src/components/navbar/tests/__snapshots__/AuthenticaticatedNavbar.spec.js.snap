// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary Navbar renders properly 1`] = `ShallowWrapper {}`;
